src/main/java/it/pagopa/dss/SignatureServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+package it.pagopa.dss;
+
+import eu.europa.esig.dss.enumerations.DigestAlgorithm;
+import eu.europa.esig.dss.model.DSSDocument;
+import eu.europa.esig.dss.model.FileDocument;
+import eu.europa.esig.dss.pades.PAdESSignatureParameters;
+import eu.europa.esig.dss.pades.SignatureFieldParameters;
+import eu.europa.esig.dss.pades.SignatureImageParameters;
+import eu.europa.esig.dss.pades.SignatureImageTextParameters;
+import eu.europa.esig.dss.pades.validation.ByteRange;
+import eu.europa.esig.dss.pdf.IPdfObjFactory;
+import eu.europa.esig.dss.pdf.PDFSignatureService;
+import eu.europa.esig.dss.pdf.ServiceLoaderPdfObjFactory;
+import eu.europa.esig.dss.pdf.pdfbox.visible.PdfBoxNativeFont;
+import eu.europa.esig.dss.spi.DSSUtils;
+import it.pagopa.dss.exception.SignatureServiceException;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.pdfbox.pdmodel.font.PDType1Font;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class SignatureServiceImpl implements SignatureServiceInterface {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SignatureServiceImpl.class);
+  private static final float DEFAULT_FONT_SIZE = 8;
+
+  @Override
+  public void generatePadesFile(File originalPdf, OutputStream outputStream)
+    throws IOException {
+    generatePadesFile(originalPdf, outputStream, null, null);
+  }
+
+  @Override
+  public void generatePadesFile(
+    File originalPdf,
+    OutputStream outputStream,
+    String signatureFieldId
+  ) throws IOException {
+    generatePadesFile(originalPdf, outputStream, signatureFieldId, null);
+  }
+
+  @Override
+  public void generatePadesFile(
+    File originalPdf,
+    OutputStream outputStream,
+    String signatureFieldId,
+    String signatureText
+  ) throws IOException {
+
+    //converting the file into a DSS document (from dss library)
+    DSSDocument documentToSign = new FileDocument(originalPdf);
+
+    //initialization of PAdES parameters
```

Comment:
this kind of comments add little value (are self-evident). instead I'd try to elaborate more ie. on what 

`parameters.setGenerateTBSWithoutCertificate(true);`

does and why it must be set to true.

Reply:
OK!